src/entry.rs

```diff
@@ -95,3 +96,15 @@ macro_rules! entry {
         }
     };
 }
+
+#[cfg(feature = "ckb-os")]
+core::arch::global_asm!(
+    ".global _start",
+    "_start:",
+    "lw a0, 0(sp)",
+    "addi a1, sp, 8",
```

Comment:
I know this code works.

But why sp needs + 8 where lw only load 4 bytes

Reply:
https://github.com/nervosnetwork/ckb-c-stdlib/blob/b51f9b1b6221398c76c6bb78d63a430d93e791bf/libc/entry.h#L19 I'm just copying the code here, I think this is due to some historical reasons: `int argc` in `int` is actually `int32_t`, but 8 bytes are allocated typically for alignment reasons.

Reply:
Yes, this should be related to the loader(how to initializing the stack).